lib/install/vchlog/vchlogger.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vchlog
+
+import (
+	"context"
+	"path"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+// VCHCreatedSignal serves as a signal struct indicating datastore folder path is available
+// Datastore: the govmomi datastore object
+// LogFileName: the filename of the destination path on datastore
+// Context: the caller context when sending the signal
+// VMPathName: the datastore path
+type VCHCreatedSignal struct {
+	Datastore   *object.Datastore
+	LogFileName string
+	Context     context.Context
+	VMPathName  string
+}
+
+// pipe: the streaming readwriter pipe to hold log messages
+var pipe *BufferedPipe
+
+// signalChan: channel for signaling when datastore folder is ready
+var signalChan chan VCHCreatedSignal
+
+// Init initializes the logger, creates the streaming pipe and makes the singaling channel.
+func Init() {
+	pipe = NewBufferedPipe()
+	signalChan = make(chan VCHCreatedSignal)
+}
+
+// Run waits until the signal arrives and uploads the streaming pipe to datastore
+func Run() {
+	sig := <-signalChan
+
+	// suffix the log file name with caller operation ID as vic-machine process identifier
+	callerOp := trace.NewOperation(sig.Context, "vic-machine")
```

Comment:
This seems like the wrong place to be calling `trace.NewOperation`. Shouldn't this code be passed a `context` where the operation has already been created? (That way the operation id in the log messages matches the operation id in the filename.)

Reply:
yes you're right. This is def not a good place to do `trace.NewOperation`...
fixing it